### DISCUSSION TITLE
Request to add Slack integration to this repo

### DISCUSSION THREAD

**Discussion by ericwmackay** [open-iscsi/open-iscsi#375] [General]
I'd like to ask if the Slack app can be added to the GitHub Integration settings for this repo.

It would add some minor conveniences including receiving Slack notifications for things like Pull Requests, new Releases, and new Issues filed rather than just emails.

**Comment by gonzoleeman** [comment]
On Sep 28, 2022, at 12:26 PM, Eric Mackay ***@***.***> wrote:
> 
> 
> I'd like to ask if the Slack app can be added to the GitHub Integration settings for this repo.
> 
> It would add some minor conveniences including receiving Slack notifications for things like Pull Requests, new Releases, and new Issues filed rather than just emails.
> 
My first thought was “why not try it out?”. :)

But I must say I’m not a fan of slack. I’d love to replace the github issues list with the google groups with “something”, but I don’t think slack is it. And I really don’t need a third way to communicate for this project (i.e. 2 are already too many).

I am forced to use slack at work, but I ignore it as much as possible. But let’s say I was willing to subscribe to an open-iscsi/github slack channel. What would it get me? I already get email when there’s a pull request or issue created or updated.

— 
Lee Duncan

**Reply by ericwmackay** [reply]
I imagine a lot of us are forced to use Slack at work :)

As far as I know this wouldn't create a Slack channel, it would just allow integration with the Github widget you can add to your Slack client. And you'd get Slack notifications (if you want) when someone creates a PR, or comments on a discussion. I think you could do basic actions like respond inline on Slack to an issue and it would post it to Github. But this wouldn't be adding any additional communication channel between users.

I see it as an alternative to email as a notification mechanism for those who want it, but otherwise it would not interfere with those who want to keep using email.

I don't think it would be any sort of replacement for Github issues or google groups, so it doesn't help out there.

**Comment by gonzoleeman** [comment]
Okay, what has to change where?

As I said, I use slack for work, but I sign in using 2FA via my work portal -- never directly to slack.

So (1) what would need to change here in the repository for open-iscsi, and (2), how would I test that?

**Comment by ericwmackay** [comment]
I think you have to first go to Slack in the GitHub marketplace: https://github.com/marketplace/slack-github

Then "Setup a Plan" (it should be free). Then the app is installed within your GitHub account. You can then select repositories that you are an Admin of, such as this one, and set permissions.

After installing the app integration, you can find it again under your account settings, under the Integration tab towards the bottom.

To test it, I made a fork of this repo, that I then enabled Slack integration on. I also installed the GitHub app within Slack (that would have to be allowed or enabled by your organization).

So:
- install the Slack integration component within GitHub
- install the GitHub integration component within Slack

Then for my forked repo, I was able to subscribe by running this command in my Slack client:

```
/github subscribe ericwmackay/open-iscsi
```

**Comment by ericwmackay** [comment]
Maybe create a private fork of this repo first, if you want to try it out and test it.

It may turn out to be less useful than I thought, and if so I wouldn't mind dropping this feature request
